### PR TITLE
Strip annotations from initializer signatures

### PR DESCRIPTION
**Pull request by josiah-wolf-oberholtzer**
In `process_signature()`, the test `getattr(obj, '__annotations__', None)` fails for classes because it tests against the class object itself and not the initializer method. This results in type annotations not being stripped from the initializer signatures.

This PR just grabs the initializer method before continuing with the test for the presence of annotations.

### CODE REVIEW

**Comment by josiah-wolf-oberholtzer** [comment]
Looks like Travis is failing due to the `backports.typing` installation, not because of the changes here.

**Comment by agronholm** [comment]
Try rebasing your code on the latest master changeset.

**Comment by agronholm** [comment]
You should also add a test that proves your patch actually fixes the problem.

**Comment by agronholm** [comment]
Why did you close the PR?

**Comment by josiah-wolf-oberholtzer** [comment]
That's odd. I didn't close it manually.

**Comment by josiah-wolf-oberholtzer** [comment]
@agronholm I've implemented explicit testing of Sphinx's output (from the text builder, for simplicity) via the `sphinx-testing` package.

Changes include:
- Adding `sphinx-testing >= 0.7.2` to the test dependencies (https://github.com/sphinx-doc/sphinx-testing)
- Moving test collateral into a dedicated `test/` directory
- Adding stub `conf.py`, `index.rst` and `dummy_module.py` files to simulate a real Sphinx installation
- Adding a test of the actual Sphinx autodoc output for the objects defined in `dummy_module.py` (see `test_sphinx_output` in the test module)

The class defined in `dummy_module.py` has normal, class and static methods. You can see in the test output that the first argument to the class and static methods is being stripped, as reported in #23. If you're happy with the way I've reorganized the tests, I can provide a fix for those missing arguments.

**Comment by josiah-wolf-oberholtzer** [comment]
Additionally, Travis is having site-wide build problems at the moment, hence the CI failures. Tests work fine locally.

**Comment by josiah-wolf-oberholtzer** [comment]
I've just gone ahead and added a fix and tests for #23 via e40d296.

**Comment by josiah-wolf-oberholtzer** [comment]
I've just found out about `sphinx.testing.fixtures`. It looks like those testing affordances supersede the older `sphinx-testing` package. I'll rework the test I added here to make use.

**Comment by agronholm** [comment]
Looks pretty good overall. Please do rename the directory to `tests` though, as this is my standard naming. There are some code changes that I would like to make too but getting the test suite to pass is the first priority.

**Comment by josiah-wolf-oberholtzer** [comment]
@agronholm Done.

**Comment by agronholm** [comment]
Thanks, I'll take another look at this later today.

File: tests/test_sphinx_autodoc_typehints.py [hunk 1]
```diff
@@ -96,3 +99,106 @@ def test_process_docstring_slot_wrapper():
...
+
+    app.build()
+
+    assert 'build succeeded' in status.getvalue()  # Build succeeded
+    assert not warning.getvalue().strip()  # No warnings
+
+    text_path = pathlib.Path(app.srcdir) / '_build' / 'text' / 'index.txt'
+    with text_path.open('r') as f:
+        text_contents = f.read().replace('–', '--')
+        text_contents == textwrap.dedent('''\
```
Comment (agronholm) [inline tests/test_sphinx_autodoc_typehints.py]: Looks like this is missing an assert...?

File: tests/test_sphinx_autodoc_typehints.py [hunk 2]
```diff
@@ -1,10 +1,13 @@
+import pathlib
 import pytest
 import sys
+import textwrap
 from typing import (
     Any, AnyStr, Callable, Dict, Generic, Mapping, Optional, Pattern, Tuple, TypeVar, Union)
 
 from sphinx_autodoc_typehints import format_annotation, process_docstring
 
+
```
Comment (agronholm) [inline tests/test_sphinx_autodoc_typehints.py]: Just FYI, the try...except import block is the fourth group of imports – no extra empty lines!

**Comment by agronholm** [comment]
It has some rough edges but it's good enough for merging and it solves a big practical problem.

**Comment by agronholm** [comment]
Thanks!

**Comment by josiah-wolf-oberholtzer** [comment]
Thanks for the merge. It's a start! This should fix #23 and #26.

**Comment by agronholm** [comment]
I took a better look at the tests you provided. The output you test against does not appear to contain any of the alterations done by sphinx-autodoc-typehints. In fact, the plugin code does not appear to run at all. I can just raise an exception right at the top of `process_signature()` and the test still happily passes.

**Comment by josiah-wolf-oberholtzer** [comment]
See #29.

The extension is running, but local changes are not visible after the initial test run. This is due to a bug in Sphinx's test fixtures.

**Pull request merged**